client/utilities/rfc-index.ts

```diff
@@ -0,0 +1,381 @@
+    .fill('')
+    .map((_, index) => ' '.repeat(index))
+
+  docListArg.sort = ['number'] // sort by earliest RFC number
+  docListArg.limit = 50 // load results in chunks of 50
+  const resultsPerPage = docListArg.limit
+
+  for (let page = 0; page <= biggestRfcNumber / resultsPerPage; page++) {
+    docListArg.offset = page * resultsPerPage
+    const response = await redApi.red.docList(docListArg)
+
+    if (abortController.signal.aborted) {
+      close()
+      return
+    }
+
+    // wait between requests so as not to overwhelm the server
+    await setTimeoutPromise(DELAY_BETWEEN_REQUESTS_MS)
+
+    if (abortController.signal.aborted) {
+      close()
+      return
+    }
+
+    push('\n\n')
+    push(
+      response.results
+        .map((result) => {
+          return [
+            padStart(result.number.toString(), longestRfcNumberLength, '0'),
+            whitespace[column1Width - longestRfcNumberLength],
+            splitWordsAt(
+              stringifyRFC(result, { longestRfcNumberLength }),
+              column2width
+            ).join(`\n${whitespace[column1Width]}`)
+          ].join('')
+        })
+        .join(' \n\n')
+    )
+  }
+
+  close()
+}
+
+const setTimeoutPromise = (timerMs: number) =>
+  new Promise((resolve) => setTimeout(resolve, timerMs))
+
+const stringifyIdentifiers = (
+  identifiers: RfcMetadata['identifiers']
+): string => {
+  if (!identifiers) return ''
+  // FIXME: when TS is fixed we should be able to delete the following line
+  const identifiersArr =
+    Array.isArray(identifiers) ? identifiers : (
+      ([identifiers] as unknown as Array<
+        NonNullable<RfcMetadata['identifiers']>
+      >)
+    )
+
+  if (identifiersArr.length === 0) return ''
+
+  return ` ${identifiersArr
+    .map(
+      (identifier) => `(${identifier.type.toUpperCase()}: ${identifier.value})`
+    )
+    .join(' ')}`
+}
+
+const stringifyAuthor = (author: RfcMetadata['authors'][number]): string => {
+  const name = author.name
+    .split(/[\s\.]/g)
+    .filter(Boolean)
+    .reduce((acc, item, index, arr) => {
+      return `${acc}${
+        index === arr.length - 1 ?
+          ` ${item}`
+        : `${item.substring(0, 1).toUpperCase()}.`
+      }`
+    }, '')
+
+  return author.affiliation === 'Editor' ? `${name}, Ed.` : name
+}
+
+type ExtraFieldNeededFormat =
+  | {
+      type: 'TXT'
+    }
+  | { type: 'HTML' }
+  | { type: 'HTMLized' }
+  | { type: 'PDF' }
+  | { type: 'PS' }
+
+type ExtraFieldNeededObsolete = {
+  number: number
+}
+
+type ExtraFieldNeededUpdate = {
+  number: number
+}
+
+type ExtraFieldNeededIsAlso = {
+  number: number
+}
+
+type ExtraFieldNeededSeeAlso = {
+  number: number
+}
+
+type ExtraFieldsNeeded = {
+  formats: ExtraFieldNeededFormat[]
+  obsoletes: ExtraFieldNeededObsolete[]
+  updates: ExtraFieldNeededUpdate[]
+  is_also: ExtraFieldNeededIsAlso[]
+  see_also: ExtraFieldNeededSeeAlso[]
+}
+
+const missingData: Record<number, Partial<RfcMetadata & ExtraFieldsNeeded>> = {
```

Comment:
fyi @jennifer-richards some data that's not present in early RFCs that would be necessary to match the existing rfc-index.txt response.

See also the `ExtraFieldsNeeded` that each RFC would also need to have.